lib/src/core/comms/secure_channel.rs

```diff
@@ -1048,8 +1071,18 @@ impl SecureChannel {
         self.local_keys.as_ref().unwrap()
     }
 
-    fn remote_keys(&self) -> &(Vec<u8>, AesKey, Vec<u8>) {
-        self.remote_keys.as_ref().unwrap()
+    fn insert_remote_keys(&mut self, keys: (Vec<u8>, AesKey, Vec<u8>)) {
+        self.remote_keys[self.token_id as usize % STORED_REMOTE_KEYS] = Some(keys);
```

Comment:
This approach, while convenient, appears to assume that `token_id`s are sequential. Is that really guaranteed?

Reply:
Yeah fair point, I guess that is not defined in the spec and so not guaranteed.